core/src/main/java/org/infinispan/statetransfer/StateProviderImpl.java

```diff
@@ -230,14 +230,13 @@ private void collectTransactionsToTransfer(Address destination,
                }
             }
          }
-         Set<Object> backupLockedKeys = tx.getBackupLockedKeys();
-         synchronized (backupLockedKeys) {
-            for (Object key : backupLockedKeys) {
-               if (segments.contains(keyPartitioner.getSegment(key))) {
-                  filteredLockedKeys.add(key);
-               }
+         //avoids the warning about synchronizing in a local variable.
+         //and allows us to change the CacheTransaction internals without having to worry about it
+         tx.forEachBackupLock(key -> {
```

Comment:
Why not `forEachLock`?

Reply:
because it isn't related to this PR (?) Do you want to replace all the `getLockedKeys()` with a `forEachLock()` ?

Reply:
Ok, I'll wait for https://issues.jboss.org/browse/ISPN-3927 then :)